shapely/strtree.py

```diff
@@ -160,45 +201,76 @@ def query(self, geom):
         >>> [o.wkt for o in tree.query(query_geom) if o.intersects(query_geom)]
         ['POINT (2 2)']
 
-        To get the original indices of the returned objects, create an
-        auxiliary dictionary. But use the geometry *ids* as keys since
-        the shapely geometry objects themselves are not hashable.
-
-        >>> index_by_id = dict((id(pt), i) for i, pt in enumerate(points))
-        >>> [(index_by_id[id(pt)], pt.wkt) for pt in tree.query(Point(2,2).buffer(1.0))]
-        [(1, 'POINT (1 1)'), (2, 'POINT (2 2)'), (3, 'POINT (3 3)')]
         """
-        if self._n_geoms == 0:
+        if self._tree is None or not self._rev:
             return []
 
         result = []
 
         def callback(item, userdata):
-            geom = ctypes.cast(item, ctypes.py_object).value
-            result.append(geom)
-
-        lgeos.GEOSSTRtree_query(self._tree_handle, geom._geom, lgeos.GEOSQueryCallback(callback), None)
+            idx = ctypes.cast(item, ctypes.py_object).value
+            result.append(idx)
 
+        lgeos.GEOSSTRtree_query(
+            self._tree, geom._geom, lgeos.GEOSQueryCallback(callback), None
+        )
         return result
 
-    def nearest(self, geom):
+    def query_geoms(self, geom: BaseGeometry) -> Sequence[BaseGeometry]:
+        """Query for nodes which intersect the geom's envelope to get
+        geometries corresponding to the items stored in the nodes.
+
+        Parameters
+        ----------
+        geom : geometry object
+            The query geometry.
+
+        Returns
+        -------
+        sequence
+            A sequence of geometry objects.
+
         """
-        Get the nearest object in the index to a geometry object.
+        items = self.query_items(geom)
+        return [self._rev[idx] for idx in items]
+
+    def query(self, geom: BaseGeometry) -> Sequence[BaseGeometry]:
+        """Query for nodes which intersect the geom's envelope to get
+        geometries corresponding to the items stored in the nodes.
+
+        This method is an alias for query_geoms. It may be removed in
+        version 2.0.
 
         Parameters
         ----------
         geom : geometry object
-            The query geometry
+            The query geometry.
 
         Returns
         -------
-        geometry object
-            The nearest geometry object in the index to `geom`.
+        sequence
+            A list or array of geometry objects.
+
+        """
+        return self.query_geoms(geom)
+
+    def nearest_item(self, geom: BaseGeometry) -> Union[int, None]:
```

Comment:
I think the return type of this function (and nearest_geom) are the only things that we'll change before 2.0. We may raise an error instead of returning None.